src/main/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsController.java

```diff
@@ -55,6 +55,19 @@ public UserCommons getUserCommonsById(
     return userCommons;
   }
 
+  // Not sure what we should put for PreAuthorize.
```

Comment:
I suggest this:

Inside the body of the controller method, check whether the user is an admin or not, and whether or not the commons has "show leaderboard" set.

The pseudocode is:

```
if (currentUser is an admin) OR (commons.getShowLeaderboard) {
   // do the things
} else {
   throw an error
}

Reply:
The exception you might consider throwing is this one.. note that the `/* user is an admin */` is pseudocode...

```java
import org.springframework.security.access.AccessDeniedException;


if (! /* user is an admin */) {
    if (! commons.getShowLeaderboard()) {
       throw new AccessDeniedException("Non admin users are not authorized to see the leaderboard in this commons");
     }
}

// now do the rest of the method...
```

Reply:
If throwing `AccessDeniedException` doesn't automatically result in a 403 status code, you can use a `ControllerAdvice` to configure that:

You can add this file under a directory called `advice`, and modify it to add a method for whatever exception you decide to throw, and what return status you want.  This example is for mapping `IllegalArgumentException` to the `BAD_REQUEST (400)` status code, but it could be used for any exception and any http return code.

```
package edu.ucsb.cs156.happiercows.advice;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestControllerAdvice;

import lombok.extern.slf4j.Slf4j;

@Slf4j
@RestControllerAdvice
public class HappierCowsControllerAdvice {

    @ExceptionHandler(IllegalArgumentException.class)
    @ResponseStatus(HttpStatus.BAD_REQUEST)
    public void handleIllegalArgumentException() {
        log.error("IllegalArgumentException thrown");
    }

}
```

Reply:
I implemented your suggestions and from the tests that I wrote, it seems to work as intended. Could you take a look to confirm?